chat-plugins/othermetas.js

```diff
@@ -147,6 +147,109 @@ exports.commands = {
 	},
 	mixandmegahelp: ["/mnm <pokemon> @ <mega stone> - Shows the Mix and Mega evolved Pokemon's type and stats."],
 
+	'!stone': true,
+	move: 'stone',
+	orb: 'stone',
+	megastone: 'stone',
+	stone: function (target) {
+		if (!this.runBroadcast()) return;
+		let targetid = toId(target);
+		if (!targetid) return this.parse('/help stone');
+		let stone;
+		if (targetid === 'dragonascent') {
+			stone = {
+				megaStone: "Rayquaza-Mega",
+				megaEvolves: "Rayquaza",
+			};
+		} else {
+			stone = Dex.getItem(target);
+		}
+		if (!stone.megaEvolves && !stone.onPrimal) return this.errorReply(`Error: Mega Stone not found.`);
+		let bannedStones = Dex.getFormat('gen7mixandmega').bannedStones;
+		if (bannedStones.includes(stone.name)) {
+			this.errorReply(`Warning: ${stone.name} is restricted to ${stone.megaEvolves} in Mix and Mega.`);
+		}
+		if (stone.isUnreleased) {
+			this.errorReply(`Warning: ${stone.name} is unreleased and is not usable in current Mix and Mega.`);
+		}
+		if (targetid === 'dragonascent') {
+			this.errorReply(`Warning: Only Pokemon with access to Dragon Ascent can mega evolve with Mega Rayquaza's traits.`);
+		}
+		// Fake Mega Stones
+		if (targetid === 'crucibellite') {
+			this.errorReply(`Warning: Crucibellite is a fake mega stone created by the CAP Project and is restricted to the CAP Crucibelle.`);
+		}
+		let baseTemplate = Dex.getTemplate(stone.megaEvolves);
+		let megaTemplate = Dex.getTemplate(stone.megaStone);
+		if (stone.id === 'redorb') { // Orbs do not have 'Item.megaStone' or 'Item.megaEvolves' properties.
+			baseTemplate = Dex.getTemplate("Groudon");
+			megaTemplate = Dex.getTemplate("Groudon-Primal");
+		} else if (stone.id === 'blueorb') {
+			baseTemplate = Dex.getTemplate("Kyogre");
+			megaTemplate = Dex.getTemplate("Kyogre-Primal");
+		}
+		let deltas = {
+			baseStats: {},
+			weightkg: megaTemplate.weightkg - baseTemplate.weightkg,
+		};
+		for (let statId in megaTemplate.baseStats) {
+			deltas.baseStats[statId] = megaTemplate.baseStats[statId] - baseTemplate.baseStats[statId];
+		}
+		if (megaTemplate.types.length > baseTemplate.types.length) {
+			deltas.type = megaTemplate.types[1];
+		} else if (megaTemplate.types.length < baseTemplate.types.length) {
+			deltas.type = baseTemplate.types[0];
+		} else if (megaTemplate.types[1] !== baseTemplate.types[1]) {
+			deltas.type = megaTemplate.types[1];
+		}
+		let details = {
+			"Gen": 6,
+			"Weight": (JSON.stringify(deltas.weightkg).startsWith("-") ? "" : "+") + Math.round(deltas.weightkg * 100) / 100 + " kg",
```

Comment:
What was wrong with `deltas.weightkg > 0`? (or `>= 0`, but personally I'd prefer "0 kg" for stones that don't change the user's weight)
Nit: weights are currently accurate to the nearest 0.1 kg